tests/migrations/test_operations.py

```diff
@@ -241,6 +241,274 @@ def test_create_model_with_db_level_fk(self):
         if connection.features.has_on_delete_db_default:
             self.assertColumnExists(f"{app_label}_rider", "pony_default_id")
 
+    def change_to_db_cascade_test(
+        self, app_label, fk_parent_model, fk_child_model, fk_fieldname, project_state
+    ):
+        operation = migrations.AlterField(
+            fk_child_model,
+            fk_fieldname,
+            models.ForeignKey(
+                f"{app_label}.{fk_parent_model}", on_delete=models.DB_CASCADE
+            ),
+        )
+        new_state = project_state.clone()
+        operation.state_forwards(app_label, new_state)
+
+        with connection.schema_editor() as editor:
+            operation.database_forwards(app_label, editor, project_state, new_state)
+
+        Rider = new_state.apps.get_model(app_label, fk_child_model)
+        Pony = new_state.apps.get_model(app_label, fk_parent_model)
+
+        pony = Pony.objects.create()
+        kwargs = {fk_fieldname: pony}
+        Rider.objects.create(**kwargs)
+
+        with self.assertNumQueries(1):
+            pony.delete()
+
+        Rider.objects.all().delete()
+        Pony.objects.all().delete()
+        with connection.schema_editor() as editor:
+            operation.database_backwards(app_label, editor, new_state, project_state)
```

Comment:
We should also test that behaviour after `database_backwards` is what we expect. For example, I'd expect the equivalent `assertNumQueries` to above to assert more than one query.

Reply:
Initially I thought of doing that, as it was already getting very long so didn't add them looking for a better loop based structure to reduce repetition. Will add after having the most optimal loop structure to write the tests.